### ISSUE TITLE
Externe Datenbank einbinden

### ISSUE THREAD

**Issue by ortwin20000** [diddipoeler/sportsmanagement#1045]
Ich habe nun eine zweite Verbandsseite für Faustball auf Joomla 3.9.16 gehoben und darauf ebenfalls JSM installiert.
Nun wollte ich die Funktion der externen Datenbank verwenden um auf meine erste Verbandsseite zugreifen zu können. Beide Verbände nutzen teilweise gemeinsame Ligen.

Ich hätte hierzu folgende Einstellung durchgeführt:
![grafik](https://user-images.githubusercontent.com/15192502/78154248-528c2f00-743c-11ea-8d09-3c68bb8168a1.png)
Wenn ich die Einstellungen so speichere erhalte ich diese Fehlermeldung:
![grafik](https://user-images.githubusercontent.com/15192502/78154347-72bbee00-743c-11ea-9802-87727bfbd328.png)
Wenn ich dann in die Konfiguration wieder aufrufen möchte erhalte ich diese:
![grafik](https://user-images.githubusercontent.com/15192502/78154427-8c5d3580-743c-11ea-92c3-ad22e446d689.png)

Ich muss somit über die Datenbank zumindest die Option ausschalten:
![grafik](https://user-images.githubusercontent.com/15192502/78154538-ab5bc780-743c-11ea-963e-feda23cf2e56.png)

Ist diese Konfiguration so dann überhaupt richtig? Wie kann ich nun auf die Daten der ersten JSM-Installation zugreifen?
Was bedeuten überhaupt diese beiden Felder ?
![grafik](https://user-images.githubusercontent.com/15192502/78154734-e5c56480-743c-11ea-9477-d8416500eccf.png)

**Comment by sigal** [comment]
Das Ganze soll mal so funktionieren, dass man sich von diddis Seite Ergebnisse abrufen und ergänzen kann. Vieleicht ein Bezahlsystem - darum die Benutzer ID. Der Server = diddis Server.

Genaues kann aber sicher diddi liefern.

**Comment by ortwin20000** [comment]
Hab mich ein wenig weiter damit beschäftigt und bin mir nicht sicher ob ein Fehler im Code vorliegt.
In der Datei administrator/components/com_sportsmanagement/helpers/sportsmanagement.php
wird in der Funktion getDBConnection folgender Code ausgeführt:
`        if ($params->get('cfg_dbprefix')) {
            $host = $config->get('host'); //replace your IP or hostname
            $user = $config->get('user'); //database user
            $password = $config->get('password'); //database password
            $database = $config->get('db'); //database name
            $prefix = $params->get('jsm_dbprefix'); //prefix if any else just give any random value
            $driver = $config->get('dbtype'); //here u can also have ms sql database driver, postgres, etc
            $debug = $config->get('config.debug');
`
Wenn ich den Teil abändere auf:
`        if ($params->get('cfg_dbprefix')) {
            $host = $params->get('jsm_host'); //replace your IP or hostname
            $user = $params->get('jsm_user'); //database user
            $password = $params->get('jsm_password'); //database password
            $database = $params->get('jsm_db'); //database name
            $prefix = $params->get('jsm_dbprefix'); //prefix if any else just give any random value
            $driver = $params->get('jsm_dbtype'); //here u can also have ms sql database driver, postgres, etc
            $debug = $config->get('config.debug');
`
funktioniert auch die Abfrage auf die andere Datenbank.

Wenn ich aber cfg_dbprefix auf "NEIN" setze wird trotzdem die externe Datenbank abgefragt.

**Comment by sigal** [comment]
Ich würde so etwas nicht öffentlich posten, da nicht so fitte user damit ihre Datenbanken unbrauchbar machen und dann das große jammern nicht aufhört.

**Comment by ortwin20000** [comment]
Naja, gibt es außer der Möglichkeit solche Fehler direkt per E-Mail zu kommunzieren eine andere Möglichkeit auf Github? So etwas wie einen geschützten Bereich?
Ich hoffe ja auch darauf, dass die Texte bis zum Schluß gelesen werden.
Da steht ja, dass nur noch die externe Datenbank abgefragt wird und somit ist meine Änderung ja nicht die Lösung.

**Comment by ortwin20000** [comment]
Ich habe jetzt ein wenig weiter gewühlt und es zumindest mal hinbekommen, dass ich bei den Views auf die externe Datenbank in der Konfiguration zugreifen kann.
Leider wirft er mir aber das Projekt beim Speichern wieder aus den Einstellungen.
Zusätzlich ist in einigen Views die Funktion für die externe Datenbank noch nicht vorhanden.
Da ich dies sehr gut für mein Projekt mit den beiden Verbänden brauchen könnte und nicht eine Multisite mit gleicher Datenbasis aufbauen möchte oder zwei JSM-Installation warten möchte, würde ich hier gerne mehr Zeit investieren und weiterhelfen. Ich bräuchte aber zumindest einige Informationen wie und wo ich hier weiter machen kann.

**Comment by diddipoeler** [comment]
@ortwin20000 Nur mal so eine Frage nebenbei. 
Möchtest Du die Komponente für kommerzielle Zwecke verwwenden ?

**Comment by ortwin20000** [comment]
Nein. Ich mache die Betreuung der Seiten ehrenamtlich aus Liebe zum Sport. Habe aber auch schon andere Projekte gemacht die vielleicht mal für kommerzielle Zwecke in Betracht gezogen werden könnten. Die sind nicht öffentlich und in meiner Freizeit entstanden.

**Comment by diddipoeler** [comment]
@ortwin20000 OK. Schon auf XING geschaut ?
Ich werde jetzt noch ein bisschen an dem Zugriff arbeiten, da ich es für
die andere Testhomepage für Joomla 4 verwenden kann.

**Comment by ortwin20000** [comment]
Ja danke. Xing hab ich gesehen 😉

**Comment by ortwin20000** [comment]
Danke für die Anpassungen. Leider funktioniert die externe Datenbank nur wenn ich sie in den Globalen Einstellungen aktiviere. Wenn ich sie nur bei den Views verwenden möchte, bekomme ich keine korrekte Auswahl der Saison. Es werden mir nur die von der eigenen Datenbank angezeigt.

**Comment by diddipoeler** [comment]
Welche Views ?

**Comment by ortwin20000** [comment]
Z.B. eventsranking oder allrounds

**Comment by ortwin20000** [comment]
Ich habe jetzt endlich den möglichen Fehler gefunden wenn ich die interne Datenbank generell verwenden möchte, aber für manche Menüeinträge doch eine externe Datenbank.
Damit kann ich zumindest mal die eigenen Ligen erfassen und zusätzlich dann vereinzelnt auch gemeinsam gespielte Ligen mit anderen Verbänden.

Mir ist aber jetzt noch aufgefallen, dass die Bilder noch nicht angezeigt werden.
Zu sehen unter http://www.faustballwien.at/feldmeisterschaft/herren/resultsranking.html
Das ist das extern eingebundene Projekt. Dort sollten eigentlich die Logos der Mannschaften angezeigt werden wie unter https://www.noefbv.at/ligaverwaltung-2/aktuelle-saison/resultsranking/0/0/136:feld-1-landesliga/0/0.html?type=0&from=0&to=0

**Comment by ortwin20000** [comment]
Ich hätte hier noch eine Frage zum Code betreffend Anbindung der externen Datenbank. Als Beispiel verwende ich hier mal die Datei view.html.php von der View ClubInfo.
In der Zeile 41 steht:
`$this->checkextrafields = sportsmanagementHelper::checkUserExtraFields('frontend', sportsmanagementModelClubInfo::$cfg_which_database);
`
auch in der Zeile 47 steht:
`$this->extrafields = sportsmanagementHelper::getUserExtraFields($this->club->id, 'frontend', sportsmanagementModelClubInfo::$cfg_which_database);
`

aber in der Zeile 91 steht:
`$this->mapconfig = sportsmanagementModelProject::getTemplateConfig('map', $this->jinput->getInt('cfg_which_database', 0));`

Alle drei Codestellen sind in der Funktion init() drin. Warum wird bei den ersten beiden das Model sportsmanagementModelClubInfo verwendet und bei der dritten Stelle dann über JInput abgefragt?

Eine Erklärung würde mir hier sehr weiterhelfen.
